### PR TITLE
Instructions Modal

### PR DESCRIPTION
**Pull request by tdaga**
Added the corn maze minigame instructions UI component. It's triggered on a random location on the map for now, but once we're able to teleport to the maze it can be switched to be on the instruction sign tile. 

instructions

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -201,6 +210,11 @@ async function GameController(initData: TownJoinResponse,
 
 function App(props: { setOnDisconnect: Dispatch<SetStateAction<Callback | undefined>> }) {
   const [appState, dispatchAppUpdate] = useReducer(appStateReducer, defaultAppState());
+  const [instructions, setShowInstructions] = useState<boolean>(appState.showInstructions);
```
Comment (lindojong) [inline frontend/src/App.tsx]: If we have `showInstructions` as part of the `appState`, do we still need this separate useState for `showInstructions`, or should we be modifying `appState` instead?

**Review by lindojong — COMMENTED** [review]
Modal looks really good! Just had one comment about the state, but otherwise looks good to me!

File: frontend/src/App.tsx (hunk 1 above)
Comment (tdaga) [reply frontend/src/App.tsx]: Good point, I've implemented this now!

**Review by kaseyl — APPROVED** [review]
Clarifying question for myself: does this make it so that you can't use arrows to close the instructions modal?
I can't figure out how to reply to this as a separate comment, but using the arrow keys still closes the modal because we set showInstructions on movement.

**Review by kaseyl — COMMENTED** [review]
Oh Instructions.tsx should have an empty line at the end of it

**Comment by kaseyl** [comment]
Can you add a line to the instructions that says "timer will start once you hit ok" or something like that?

**Review by lindojong — APPROVED** [review]

**Pull request merged**